Update pango to 1.50.7

cbouss opened the pull request:
home: https://pango.gnome.org/
doc_url: https://docs.gtk.org/Pango/
dev_url: https://gitlab.gnome.org/GNOME/pango/-/tree/1.50.7
changelog: https://gitlab.gnome.org/GNOME/pango/-/blob/1.50.7/NEWS
compare: https://gitlab.gnome.org/GNOME/pango/-/compare/1.48.7...1.50.7?from_project_id=1515
Meson:
- https://gitlab.gnome.org/GNOME/pango/-/blob/1.50.7/meson_options.txt
- https://gitlab.gnome.org/GNOME/pango/-/blob/1.50.7/meson.build

Jira: https://anaconda.atlassian.net/browse/DSNC-4873

Changes:
- Update to 1.50.7
- Add Windows build
- Update about section
- Add tests in test section

Note: Library tests results in build files are ignored because some are failing. This was already the case in previous 1.48.7.

cbouss commented:
TODO once approved:
- after harfbuzz is merged, remove abs.yaml and conda_config_build.yaml, then merge.

varlackc reviewed (commented):
we might need to adjust the pinnings for `harfbuzz` to be at least `2.6.0`
https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L231

might be a good idea to specify the required version of `glib`
https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L228

we might want to specify the recommended version for `fribidi`
https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L229

The `libthai` package seems to be a requirement, might have to see if this is needed for the package to work properly
https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L230

might have to specify the `fontconfig` recommended versions
https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L232

The `xft` package seems to be a required package
https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L233

The package `Cairo` seems to have a minimum version recomended
https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L234

there might be some reasons to diverge from the recommendations, but I think that it might be worth while investigating into this before merging.

cbouss commented:
> we might need to adjust the pinnings for `harfbuzz` to be at least `2.6.0` https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L231
> 
> might be a good idea to specify the required version of `glib` https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L228
> 
> we might want to specify the recommended version for `fribidi` https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L229
> 
> The `libthai` package seems to be a requirement, might have to see if this is needed for the package to work properly https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L230
> 
> might have to specify the `fontconfig` recommended versions https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L232
> 
> The `xft` package seems to be a required package https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L233
> 
> The package `Cairo` seems to have a minimum version recomended https://gitlab.gnome.org/GNOME/pango/-/blob/a2ccd36a42e039d3600b04fe37fdc47f267d90c7/meson.build#L234
> 
> there might be some reasons to diverge from the recommendations, but I think that it might be worth while investigating into this before merging.

Thanks for the advice. 
I added the run section which was missing from the recipe and that I hadn't spotted before.
I pinned harfbuzz to 4.3.0 in the cbc. Once all that depend on harfbuzz are rebuilt, I will update aggregate cbc and merge.
Likewise for cairo which will have it's pinning set to 1.16.
libthai and xft are optional. They are not included in this build.
I updated the minimum requirements of glib, fribidi and fontconfig (FYI glib and fontconfig also have pinning set in aggregate cbc).

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -30,33 +33,56 @@ requirements:
-    - {{ cdt('xorg-x11-proto-devel') }}   # [linux and s390x]
-    - make                                # [unix]
+    - {{ cdt('xorg-x11-proto-devel') }}   # [linux and (s390x or ppc64le or x86_64)]; needed for cairo support
     - meson
```
The minimum required version of `meson` is [0.55.3](https://gitlab.gnome.org/GNOME/pango/-/blob/1.50.7/meson.build#L11) and https://gitlab.gnome.org/GNOME/pango/-/blob/1.50.7/NEWS#L202
